### PR TITLE
Use TSX parser instead of Typescript parser for JavaScript and JSX

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
JavaScript (and `.jsx`) files can contain JSX (and cannot use TypeScript's type assertion syntax), so we need to use the TSX dialect of TypeScript when parsing and analyzing them.

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
Is there any testing that should be added for this? Previously, we would have gotten parse errors on any `.js` files that used JSX, so maybe just a unit test that we can parse a `.js` file with some JSX in it?

**Comment by tclem** [comment]
Let's do something like [this](https://github.com/github/semantic/blob/a6c073895cae7718113197ca0c8db2b2b7286f6f/test/fixtures/tsx/corpus/jsx-elements.A.tsx), but with a `.jsx` extension. You can use `script/generate-example` to help bootstrap the expected output.

**Comment by maxbrunsfeld** [comment]
While we were here, @rewinfrey and I also added some limited support for Type arguments in TSX (https://github.com/Microsoft/TypeScript/pull/22415).

File: test/fixtures/tsx/corpus/jsx-elements.diffA-B.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@
       (JsxElement
         (JsxOpeningElement
           (Identifier)
+          (Empty)
```
Comment (maxbrunsfeld) [inline test/fixtures/tsx/corpus/jsx-elements.diffA-B.txt]: `JsxOpeningElement` terms now have an optional `TypeArguments` child.

**Comment by erickadalberto** [comment]
> JavaScript files can contain JSX (and cannot use TypeScriptyp so we need to use *the TSX dialect of TypeScript when parsing and analyzing them\

**Pull request merged**